adserver/views.py

```diff
@@ -956,6 +958,110 @@ def get_context_data(self, **kwargs):
         return context
 
 
+class AdvertiserAuthorizedUsersView(
+    AdvertiserAccessMixin, UserPassesTestMixin, ListView
+):
+
+    """Authorized users for an advertiser."""
+
+    model = get_user_model()
+    template_name = "adserver/advertiser/users.html"
+
+    def get_context_data(self, **kwargs):  # pylint: disable=arguments-differ
+        context = super().get_context_data(**kwargs)
+        context.update({"advertiser": self.advertiser, "users": self.object_list})
+        return context
+
+    def get_queryset(self):
+        self.advertiser = get_object_or_404(
+            Advertiser, slug=self.kwargs.get("advertiser_slug", "")
+        )
+        return self.advertiser.user_set.all()
+
+
+class AdvertiserAuthorizedUsersInviteView(
+    AdvertiserAccessMixin, UserPassesTestMixin, CreateView
+):
+
+    """Invite additional authorized users for an advertiser."""
+
+    form_class = InviteUserForm
+    model = get_user_model()
+    template_name = "adserver/advertiser/users-invite.html"
+
+    def dispatch(self, request, *args, **kwargs):
+        self.advertiser = get_object_or_404(
+            Advertiser, slug=self.kwargs["advertiser_slug"]
+        )
+        return super().dispatch(request, *args, **kwargs)
+
+    def form_valid(self, form):
+        result = super().form_valid(form)
+        self.object.advertisers.add(self.advertiser)
+        messages.success(
+            self.request,
+            _("Successfully invited %(user)s to %(advertiser)s")
+            % {"user": self.object.email, "advertiser": self.advertiser},
+        )
```

Comment:
Are there any error cases here? I guess most (eg. invalid email) would be handled in the form.

Reply:
Ya, the only error case is an invalid email.